### PR TITLE
fix: correct CountWildcardRule and move analyzer into a new directory.

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Closes #.

# Rationale for this change

related PR: #5627

# What changes are included in this PR?

 - move analyzer to new dir 
 - fix CountWildcardRule.

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Review by yjshen — APPROVED** [review]

**Comment by mingmwang** [comment]
Should this rule internally call `transform_up() ` or `transform_down()` ?

**Comment by jackwener** [comment]
> Should this rule internally call `transform_up() ` or `transform_down()` ?

Oh, this Rule don't handle it ...... I will correct it.

File: datafusion/core/tests/dataframe.rs [hunk 1]
```diff
@@ -51,16 +51,18 @@ async fn count_wildcard() -> Result<()> {
     let sql_results = ctx
         .sql("select count(*) from alltypes_tiny_pages")
         .await?
+        .select(vec![count(Expr::Wildcard)])?
         .explain(false, false)?
         .collect()
         .await?;
 
+    // add `.select(vec![count(Expr::Wildcard)])?` to make sure we can analyze all node instead of just top node.
```
Comment (alamb) [inline datafusion/core/tests/dataframe.rs:59]: 👍

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @jackwener

**Comment by jiangzhx** [comment]
thanks @jackwener for correcting it, I learned a lot from your code。

**Pull request merged**